tycho_simulation_py/python/tycho_simulation_py/evm/pool_state.py

```diff
@@ -97,8 +98,8 @@ def __init__(
         self.involved_contracts: set[Address] = involved_contracts or set()
         """A set of all contract addresses involved in the simulation of this pool."""
 
-        self.token_storage_slots: dict[Address, tuple[int, int]] = (
-            token_storage_slots or {}
+        self.token_storage_slots: dict[Address, tuple[tuple[int, int], ContractCompiler]] = (
```

Comment:
Can you mention a bit why we have this here per contract in the docs? (I think it makes a lot of sense)